### PR TITLE
Max/sqlerror codes

### PR DESCRIPTION
**Pull request by max-hoffman**
Append error codes/sql state to error messages for downstream clients.

### CODE REVIEW

File: server/handler.go [hunk 1]
```diff
@@ -321,7 +321,7 @@ func (h *Handler) doQuery(
 	}()
 	if err != nil {
 		logrus.Tracef("Error running query %s: %s", query, err)
-		return err
+		return sql.CastSQLError(err)
```
Comment (zachmu) [inline server/handler.go:324]: You need this in more places. Check out the other return code paths.

Or to make it easier, you might wrap the invocation of this entire function in another that inspects the error as necessary.

File: sql/errors.go [hunk 2]
```diff
@@ -127,3 +130,15 @@ var (
 	// ErrTruncateReferencedFromForeignKey is returned when a table is referenced in a foreign key and TRUNCATE is called on it.
 	ErrTruncateReferencedFromForeignKey = errors.NewKind("cannot truncate table %s as it is referenced in foreign key %s on table %s")
 )
+
+func CastSQLError(err error) *mysql.SQLError {
+	var code int
+	var sqlState string = ""
+	switch {
+	case ErrTableNotFound.Is(err):
+		code = 1146
```
Comment (zachmu) [inline sql/errors.go:139]: Probably better to use the constants for these that vitess defines. They can be exported if they aren't already. Hard to review changes here with them being magic numbers.

**Review by zachmu — COMMENTED** [review]
Good start, not complete

File: server/handler.go [hunk 3]
```diff
@@ -156,7 +159,10 @@ func (h *Handler) ComQuery(
 	query string,
 	callback func(*sqltypes.Result) error,
 ) error {
-	return h.doQuery(c, query, nil, callback)
+	if err := h.doQuery(c, query, nil, callback); err != nil {
```
Comment (zachmu) [inline server/handler.go:162]: Nit: just make CastSQLError handle the nil err case and you don't need the conditional logic

File: server/handler.go (hunk 1 above)
Comment (zachmu) [reply server/handler.go:330]: Remove now?

**Review by zachmu — APPROVED** [review]
LG

File: server/handler.go (hunk 1 above)
Comment (max-hoffman) [reply server/handler.go]: ended up adding a new function because of go's type/value interface nil idiosyncrasies

**Pull request merged**